src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -64,10 +69,75 @@ const TaxonomyListHeaderButtons = () => {
   );
 };
 
+const OrganizationFilterSelector = ({
+  isOrganizationListLoaded,
+  organizationListData,
+  selectedOrgFilter,
+  setSelectedOrgFilter,
+}) => {
+  const intl = useIntl();
+  const isOrgSelected = (value) => (value === selectedOrgFilter ? <Check /> : null);
+  const selectOptions = [
+    <MenuItem
+      key="all-orgs-taxonomies"
+      className="x-small"
+      iconAfter={() => isOrgSelected('All taxonomies')}
+      onClick={() => setSelectedOrgFilter('All taxonomies')}
+    >
+      { isOrgSelected('All taxonomies')
+        ? intl.formatMessage(messages.orgInputSelectDefaultValue)
+        : intl.formatMessage(messages.orgAllValue)}
+    </MenuItem>,
+    <MenuItem
+      key="unassigned-taxonomies"
+      className="x-small"
+      iconAfter={() => isOrgSelected('Unassigned')}
+      onClick={() => setSelectedOrgFilter('Unassigned')}
+    >
+      { intl.formatMessage(messages.orgUnassignedValue) }
+    </MenuItem>,
+  ];
+
+  if (isOrganizationListLoaded && organizationListData) {
+    organizationListData.map(org => (
```

Comment:
You're not really mapping values, just iterating over them. 
```suggestion
    organizationListData.forEach(org => (
```